### PR TITLE
updated packages, updated .eslintrc

### PR DESCRIPTION
**Pull request by manonthemat**

### CODE REVIEW

**Comment by fatso83** [comment]
We'll need to update the travis config to run on later Node versions with ES6 support for this to work.

File: .eslintrc [hunk 1]
```diff
@@ -25,7 +25,6 @@
         "no-else-return": 2,
         "no-empty": 2,
         "no-empty-character-class": 2,
-        "no-empty-label": 2,
```
Comment (fearphage) [inline .eslintrc]: This looks like it was replaced by http://eslint.org/docs/rules/no-unused-labels

File: .eslintrc [hunk 2]
```diff
@@ -134,6 +133,7 @@
         "indent": [2, 4, {"SwitchCase": 1}],
         "init-declarations": 0,
         "key-spacing": [2, { "beforeColon": false, "afterColon": true }],
+        "keyword-spacing": ["error", { "before": true }],
```
Comment (fearphage) [inline .eslintrc]: That's the default value. `"keyword-spacing": "error"` should suffice.

**Comment by fearphage** [comment]
> We'll need to update the travis config to run on later Node versions with ES6 support for this to work.

We should do that anyway, but an alternative solution is to add `node_modules/` to the `.eslintignore`. /cc @manonthemat.

File: .eslintrc [hunk 3]
```diff
@@ -134,6 +134,7 @@
         "indent": [2, 4, {"SwitchCase": 1}],
         "init-declarations": 0,
         "key-spacing": [2, { "beforeColon": false, "afterColon": true }],
+        "keyword-spacing": ["error"],
```
Comment (fearphage) [inline .eslintrc]: Minor: Remove the braces.

**Comment by fearphage** [comment]
@manonthemat To get the tests passing, you'll need to add a newer version of nodejs to Travis and switch the eslint task to use the newer version. `v6.7.0` is the latest.

I'm available to help if you'd like, but don't let me get in your way.

**Comment by fearphage** [comment]
Also thanks for helping out! :+1:

**Comment by manonthemat** [comment]
I just added nodejs 6.7 to travis. How do I switch the eslint task to use the newer version?
I apologize for these many commits and the help that is required for this PR to get through. I appreciate the help.

**Comment by fearphage** [comment]
> How do I switch the eslint task to use the newer version?

This is the line you need to change: https://github.com/sinonjs/sinon-test/pull/27/files#diff-354f30a63fb0907d4ad57269548329e3L22

It's a little hard to read, but you need to replace `0.12` (that's the node version) with your newly added `6.7` version. It'll be "x6.7" when you're done.

## Backstory and Details

That is just a line of bash scripting. It reads:

If the `$TRAVIS_NODE_VERSION` is `0.12`, run `eslint .`.

If you felt like it, you could remove the `x` padding on both sides of the equation. You don't have to though. The way Travis CI works is it runs the tests N times where N is the number of versions listed under `node_js:` in the yaml file. So each time it runs, it sets that environment variable so you'll know what version is active. Your change will make this only run in the latest version of node. 

> I apologize for these many commits and the help that is required for this PR to get through.

I don't care if it takes a hundred commits. We appreciate your contribution!

**Comment by fearphage** [comment]
@manonthemat Thanks again!

**Comment by fearphage** [comment]
@sinonjs/sinon-test-maintainers FYI I had to clear the travis build cache and re-run the build to get the tests passing.

**Pull request merged**